util/stmtsummary/statement_summary.go

```diff
@@ -578,19 +579,25 @@ func (ssbd *stmtSummaryByDigest) toCurrentDatum(beginTimeForCurInterval int64) [
 
 	// `ssElement` is lazy expired, so expired elements could also be read.
 	// `beginTime` won't change since `ssElement` is created, so locking is not needed here.
-	if ssElement == nil || ssElement.beginTime < beginTimeForCurInterval {
+	if user == nil {
+		return ssElement.toDatum(ssbd)
+	}
+	if ssElement == nil || ssElement.beginTime < beginTimeForCurInterval || user == nil ||
```

Comment:
What will happen if `user` is `nil`?
Line 582 says it will return all data, but line 585 says it will return nil.
BTW, line 582 didn't check whether `ssElement` is `nil`.

Reply:
Some tests in infoschema will pass user as nil. In previous commit, it caused the tests failed.

https://github.com/pingcap/tidb/pull/15184/commits/966267c72d3638880fd8a6bfd9b3b87be56a1113

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/26331/pipeline

Line 582 is a mistake. Fixed.